merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -484,9 +484,15 @@ public SimulationResults computeResults(
             serializeReturnValue(directive, e.returnValue())
         ));
       } else {
-        unsimulatedActivities.put(activityId, new SerializedActivity(
+        final var activity = new SerializedActivity(
             directive.getType(),
-            directive.getArguments()));
+            directive.getArguments());
+
+        unsimulatedActivities.put(activityId, activity); // Record this activity as unfinished
+
+        if (state instanceof ExecutionState.IllegalSource<?> e) { // Record this activity as unconstructable
+          unconstructableActivities.put(activityId, new SerializedActivity.Unconstructable(activity, e.reason()));
```

Comment:
I'm considering an unconstructable activity to also be an unsimulated ("unfinished") activity. I guess it's technically correct that an unconstructable activity is unsimulated/unfinished, but maybe we want to keep these as distinct sets of activities?